### ISSUE TITLE
KinitoPET (2075070)

### ISSUE THREAD

**Issue by IWarnedYouAboutStairs** [ValveSoftware/Proton#7406]
# Compatibility Report
- Name of the game with compatibility issues: KinitoPET
- Steam AppID of the game: 2075070

## System Information
- GPU: RTX 3070
- Video driver version: 545.29.06
- Kernel version: 6.6.10
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/IWarnedYouAboutStairs/a6a21bcf3d3d7f8c8940b3894220ac75
- Proton version: Experimental, 8.0-4

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-2075070.tar.gz](https://github.com/ValveSoftware/Proton/files/13930913/steam-2075070.tar.gz)

## Symptoms 
The game does not start. Sometimes a black window appears for a split second before crashing, and sometimes nothing appears at all.

Curiously, the game seems to start up and work fine when running the executable directly under the standard system Wine installation (wine-8.21).

## Reproduction
Try to run the game as normal. The issue persists under Proton 8.0-4, Proton Experimental, and GE-Proton8-27, as well as under both X11 and Wayland.
As mentioned earlier, the game runs as expected when using system Wine to run the executable directly from the install directory.

**Comment by kisak-valve** [comment]
Hello @IWarnedYouAboutStairs, these look like some lines of interest from your Proton log:
```
Wallpapaer.tscn - WALLPAPER COULD NOT BE LOADED: ERROR
[...]
Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at DESKTOPICONS.isVisible () [0x0000f] in <e7d18c6f1562477b8ece06c751629070>:0
```

**Comment by IWarnedYouAboutStairs** [comment]
I was going to file a bug report to the developer originally, but the game seems to start and play without issue through Wine, suggesting it's a Proton issue rather than an issue with the game itself. I recorded a video to demonstrate: https://streamable.com/qhbk5z

**Comment by lambdadeltakay** [comment]
The game seems to startup like half the time correctly if i run in virtual desktop

**Comment by kisak-valve** [comment]
KinitoPET is unplayable on Proton

Issue transferred from https://github.com/ValveSoftware/Proton/issues/7739.
@robstarmcdonald posted on 2024-05-17T21:08:32:

# System Details Report

## Hardware Information:
- **Hardware Model:**                              MSI MS-7984
- **Memory:**                                      32.0 GiB
- **Processor:**                                   Intel® Core™ i7-6700K × 8
- **Graphics:**                                    AMD Radeon™ RX 6650 XT
- **Disk Capacity:**                               4.2 TB

## Software Information:
- **Firmware Version:**                            1.40
- **OS Name:**                                     Ubuntu 24.04 LTS
- **OS Build:**                                    (null)
- **OS Type:**                                     64-bit
- **GNOME Version:**                               46
- **Windowing System:**                            X11
- **Kernel Version:**                              Linux 6.8.0-31-generic
- **Proton version:**                             Proton Experimental (9.0)

## I confirm:
- [ ] I haven't found an existing compatibility report for this game.
- [x] I have checked whether there are updates for my system available.
***
@robstarmcdonald commented on 2024-05-17T21:09:12:

According to PROTON_LOG=1 %command%

`<Proton log omitted, see linked issue report>`

**Comment by robstarmcdonald** [comment]
According to PROTON_LOG=1 %command%

[steam-2075070.log](https://github.com/ValveSoftware/Proton/files/15363321/steam-2075070.log)

**Comment by simifor** [comment]
The game should now launch on proton experimental bleeding edge

**Comment by SAndrass12** [comment]
I found the site [kinitopet](https://kinitopetgame.com), is this an original game or not?

**Comment by robstarmcdonald** [comment]
Yep, I'd tested it on Proton Experimental on bleeding-edge mode, it runs
fine.

**Comment by silverhadch** [comment]
It doesnt launch on Experimental either...

**Comment by robstarmcdonald** [comment]
> It doesnt launch on Experimental either...

Correct @silverhadch, because it used to, but now, it doesn't work on Proton Experimental anymore. You're not alone.

Also, did you tried bleeding-edge for that version?

**Comment by silverhadch** [comment]
@robstarmcdonald 
Yes i run the Arch Version. It might be a regression in Proton.

**Comment by kisak-valve** [comment]
Hello @silverhadch, @robstarmcdonald, can one of you please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.) Also, please copy the contents of Steam Runtime Diagnostics from Steam (`Steam` -> `Help` -> `Steam Runtime Diagnostics`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by silverhadch** [comment]
[steam-2075070.log](https://github.com/user-attachments/files/18546387/steam-2075070.log)

**Comment by silverhadch** [comment]
@kisak-valve
https://gist.github.com/silverhadch/1bb9cf6e8450fca48ed8193773fa8440
Cant currently run the application. @robstarmcdonald it would be nice if you provide a log.

**Comment by robstarmcdonald** [comment]
Hello @kisak-valve and @silverhadch,

Thanks for both of your requests, here's what I got for you two.
[steam-2075070.log](https://github.com/user-attachments/files/18556190/steam-2075070.log)

Yes, with [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) on Advanced Launch Options. [Steam Snap](https://snapcraft.io/steam) should have similar compatibility like a Deb version when it comes to [Proton.](https://github.com/ValveSoftware/Proton)

Also @silverhadch, if the game crashes on a load while you added `PROTON_LOG=1 %command%`, it should create a log file `steam-"appid".log` on a home page or `/home/"username"/snap/steam/common` when it comes to [Steam Snap.](https://snapcraft.io/steam)

**Comment by silverhadch** [comment]
@kisak-valve @robstarmcdonald 
Okay maybe its a bit late but i think i got a lead. When installing Windows in an VM i dont have an GPU so OpenGL Drivers etc. wont work and launching the game fails with an error message. I used an precompiled Mesa with newer OpenGL from https://github.com/pal1000/mesa-dist-win/releases. The Application launches but... it dies the exact same way in Proton every Frame the same as if i am trying to run the Game on Linux via Proton. Blackscreening close balckscreening close tiny black box in corner then crash. So it might be a problem with the newer OpenGL Version? But just a hunch.
